### ISSUE TITLE
`otel-cli exec` doesn't successfully parse a result from an HTTP service

### ISSUE THREAD

**Issue by parsonsmatt** [equinix-labs/otel-cli#262]
While diagnosing how to make trapping signals work in Go, I identified that using an HTTP client has a bug. The [call to `processHTTPStatus`](https://github.com/equinix-labs/otel-cli/blob/5f1223a809007b088af2ea9b9c534a798589f1f4/otlpclient/otlp_client_http.go#L83) here goes on to do [`proto.Unmarshal`](https://github.com/equinix-labs/otel-cli/blob/5f1223a809007b088af2ea9b9c534a798589f1f4/otlpclient/otlp_client_http.go#L95). When communicating with Honeycomb, this causes an error, because Honeycomb returns JSON.

Debug output:

```
guess we're almost done. resp:  &{200 OK 200 HTTP/1.1 1 1 map[Access-Control-Allow-Origin:[*] Connection:[keep-alive] Content-Length:[2] Content-Type:[application/json] Date:[Fri, 08 Sep 2023 02:16:22 GMT] Vary:[Accept-Encoding]] 0xc000674300 2 [] false false map[] 0xc00015e400 0xc000197a20}
```

Specifically `Content-Type:[application/json]`.

Changing `proto.Unmarshal` to `json.Unmarshal` appears to fix the issue for the Honeycomb case, but i suspect that you'd want to use the content-type to determine the exact unmarshal function.

**Comment by tobert** [comment]
Oh, fascinating. I don't think the spec says this is illegal, but it does say the response should be another protobuf, not json. So I think this is a Honeycomb bug that the otel stack usually ignores.

I'm open to having otel-cli switch on content type and handle both, and will also ping HC folks to see if they want to fix this.

**Comment by tobert** [comment]
@parsonsmatt I merged #263 so otel-cli should return a better error. I spoke with someone at Honeycomb and they felt this was the better option than accepting the JSON.

**Comment by tobert** [comment]
Going to close this. Please reopen if the chosen strategy prevents you from solving problems.
